src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -103,7 +106,30 @@ public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exc
     }
 
     @Test
-    public void parseLocation_null_throwsNullPointerException() {
+    public void parseGoals_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseGoals((String) null));
+    }
+
+    @Test
+    public void parseGoals_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseGoals(INVALID_GOALS));
+    }
+
+    @Test
+    public void parseGoals_validValueWithoutWhitespace_returnsGoals() throws Exception {
+        Goals expectedGoals = new Goals(VALID_GOALS);
+        assertEquals(expectedGoals, ParserUtil.parseGoals(VALID_GOALS));
+    }
+
+    @Test
+    public void parseGoals_validValueWithWhitespace_returnsTrimmedGoals() throws Exception {
+        String goalsWithWhitespace = WHITESPACE + VALID_GOALS + WHITESPACE;
+        Goals expectedGoals = new Goals(VALID_GOALS);
+        assertEquals(expectedGoals, ParserUtil.parseGoals(goalsWithWhitespace));
+    }
+
+    @Test
+    public void parseAddress_null_throwsNullPointerException() {
```

Comment:
```suggestion
    public void parseLocation_null_throwsNullPointerException() {
```
Perhaps change to Location